### PR TITLE
Enhance grant permission on topic

### PR DESCRIPTION
**Pull request by TakaHiR07**
### Motivation
related to https://github.com/apache/pulsar/issues/16768

### Modifications
no need to grant permission on each partition of partitioned-topic

### Does this pull request potentially affect one of the following parts:

*If `yes` was chosen, please highlight the changes*

  - Dependencies (does it add or upgrade a dependency): no
  - The public API: no
  - The schema: no
  - The default values of configurations: no
  - The wire protocol: no
  - The rest endpoints: no
  - The admin cli options: no
  - Anything that affects deployment: no

### Documentation

Check the box below or label this PR directly.

Need to update docs? 

- [ ] `doc-required` 
(Your PR needs to update docs and you will update later)
  
- [x] `doc-not-needed` 
(Please explain why)
  
- [ ] `doc` 
(Your PR contains doc changes)

- [ ] `doc-complete`
(Docs have been already added)

### CODE REVIEW

**Comment by Technoboy-** [comment]
@TakaHiR07 Should you commit to the master branch?

File: pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/impl/PersistentTopicsBase.java [hunk 1]
```diff
@@ -283,14 +283,6 @@ protected void internalGrantPermissionsOnTopic(String role, Set<AuthAction> acti
...
 
-        PartitionedTopicMetadata meta = getPartitionedTopicMetadata(topicName, true, false);
-        int numPartitions = meta.partitions;
-        if (numPartitions > 0) {
-            for (int i = 0; i < numPartitions; i++) {
-                TopicName topicNamePartition = topicName.getPartition(i);
-                grantPermissions(topicNamePartition.toString(), role, actions);
-            }
-        }
         grantPermissions(topicName.toString(), role, actions);
```
Comment (nodece) [inline pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/impl/PersistentTopicsBase.java:286]:
```suggestion
        grantPermissions(topicName.getPartitionedTopicName(), role, actions);
```

**Comment by nodece** [comment]
I think you should commit your PR to the master branch.

**Comment by TakaHiR07** [comment]
> I think you should commit your PR to the master branch.

@nodece @Technoboy- I have found that revoke permission also has problem. I will open an other pr soon to fix this problem. This pr would be close

**Pull request closed without merging**